Decred Wallet/Custom Views/FloatingLabelTextInput.swift

```diff
@@ -0,0 +1,112 @@
+//
+//  FloatingLabelTextInput.swift
+//  Decred Wallet
+//
+// Copyright (c) 2018-2019 The Decred developers
+// Use of this source code is governed by an ISC
+// license that can be found in the LICENSE file.
+
+import UIKit
+
+class FloatingLabelTextInput: UITextField {
+    var border = CALayer()
+    var floatingLabel: UILabel!
+    var floatingLabelHeight: CGFloat = 14
+    var button = UIButton(type: .custom)
+    
+    @IBInspectable
+    var _placeholder: String?
+    
+    @IBInspectable
+    var floatingLabelColor: UIColor = UIColor.appColors.decredBlue {
+        didSet {
+            self.floatingLabel.textColor = floatingLabelColor
+            self.setNeedsDisplay()
+        }
+    }
+    
+    @IBInspectable
+    var floatingLabelBackground: UIColor = UIColor.white.withAlphaComponent(1) {
+        didSet {
+            self.floatingLabel.backgroundColor = floatingLabelBackground
+            self.setNeedsDisplay()
+        }
+    }
+    
+    @IBInspectable
+    var floatingLabelFont: UIFont = UIFont(name: "Source Sans Pro", size: 14)! {
+        didSet {
+            self.floatingLabel.font = floatingLabelFont
+        }
+    }
+    
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.floatingLabel = UILabel(frame: CGRect.zero)
+        self._placeholder = (self._placeholder != nil) ? self._placeholder : placeholder
+        layer.borderWidth = 0.8
+        borderStyle = .roundedRect;
+        
+        self.addTarget(self, action: #selector(self.addFloatingLabel), for: .editingDidBegin)
+        self.addTarget(self, action: #selector(self.removeFloatingLabel), for: .editingDidEnd)
+    }
+    
+    // Add a floating label to the view on becoming first responder
+    @objc func addFloatingLabel() {
+        if self.text == "" {
+            self.floatingLabel.textColor = floatingLabelColor
+            self.floatingLabel.font = floatingLabelFont
+            self.floatingLabel.text = self._placeholder
```

Comment:
I see `self._placeholder` is assigned a value in init. Why not add a `didSet` property to the `_placeholder` variable so that changing the variable's value auto sets the new value to `self.floatingLabel.text`. That would render this code here unnecessary and simplify this `addFloatingLabel` method.